main/src/ca/uwaterloo/flix/api/Flix.scala

```diff
@@ -213,16 +213,11 @@ class Flix {
     Compiler.compile(getSourceInputs, hooks.toMap)(genSym)
   }
 
-  /**
-    * Solves the Flix program and returns the minimal model.
-    *
-    * NB: Automatically calls `compile()` thus there is no reason to do so manually.
-    */
-  def solve(): Validation[Model, CompilationError] = {
+  // TODO: Refactor this into compile()
```

Comment:
Yes. We need to come up with a general mechanism for the compiler phases. Furthermore, compile() is used in hundreds of test cases, so we want to get the signature right, before starting on a long refactoring process.